File: backend/app/schemas/discovered_job.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field

from app.models.discovered_job import EligibilityStatus


class DiscoveredJobResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    title: str
    company: str
    location: str | None
    description: str | None
    employment_type: str | None
    source: str
    source_url: str | None
    remote_status: EligibilityStatus
    visa_sponsorship_status: EligibilityStatus
    relocation_status: EligibilityStatus
    international_eligibility_status: EligibilityStatus
    discovered_at: datetime
    salary: str | None
    experience_level: str | None
    skills: str | None
    created_at: datetime
    updated_at: datetime


class DiscoveredJobListResponse(BaseModel):
    items: list[DiscoveredJobResponse]
    page: int
    page_size: int
    total: int
    pages: int

File: frontend/src/types/discoveredJob.ts
export type EligibilityStatus = 'UNKNOWN' | 'YES' | 'NO'

export const ELIGIBILITY_STATUS_ORDER: EligibilityStatus[] = [
  'UNKNOWN',
  'YES',
  'NO',
]

export const ELIGIBILITY_STATUS_LABELS: Record<EligibilityStatus, string> = {
  UNKNOWN: 'Unknown',
  YES: 'Yes',
  NO: 'No',
}

export type DiscoveredJob = {
  id: number
  title: string
  company: string
  location: string | null
  description: string | null
  employment_type: string | null
  source: string
  source_url: string | null
  remote_status: EligibilityStatus
  visa_sponsorship_status: EligibilityStatus
  relocation_status: EligibilityStatus
  international_eligibility_status: EligibilityStatus
  discovered_at: string
  salary: string | null
  experience_level: string | null
  skills: string | null
  created_at: string
  updated_at: string
}

export type DiscoveredJobListResponse = {
  items: DiscoveredJob[]
  page: number
  page_size: number
  total: number
  pages: number
}

export type DiscoveredJobSortBy = 'discovered_at' | 'company' | 'title'
export type DiscoveredJobSortOrder = 'asc' | 'desc'

export type DiscoveredJobListParams = {
  search?: string
  location?: string
  remote_status?: EligibilityStatus
  visa_sponsorship_status?: EligibilityStatus
  employment_type?: string
  page?: number
  page_size?: number
  sort_by?: DiscoveredJobSortBy
  sort_order?: DiscoveredJobSortOrder
}
